sdk/log/simple.go

```diff
@@ -17,12 +17,6 @@ type SimpleProcessor struct {
 }
 
 // NewSimpleProcessor is a simple Processor adapter.
-//
-// This Processor is not recommended for production use. The synchronous
-// nature of this Processor make it good for testing, debugging, or
-// showing examples of other features, but it can be slow and have a high
-// computation resource usage overhead. [NewBatchProcessor] is recommended
-// for production use instead.
```

Comment:
Won't this differ from the trace SDK?

Reply:
Do you support making the same change in Trace SDK? If so would you prefer to have this in this or separate PR?

Reply:
I would prefer we continue with past decisions that we have made, published, and promoted. The common use case should not use this processor for both local and remote resource consumption issues. Warning users about these still seems relevant. I would prefer we remain consistent with the tracing signal unless there is reason for the common use case to diverge.

Reply:
I think it depends on the exporter whether it is safe or not to be used with simple processor. 

I agree that probably for most exporters it would be better to use the batching processor.

The alternative could be that some exporters like ETW, user_events may explicitly that they are safe to be used with simple processor. 

WDYT?

Reply:
Maybe something like "check with your exporter before using this"(?)

Reply:
Gotcha. I will try to come up with something tomorrow. Thanks 👍

Reply:
PTAL ef5fe1b4

I can update the comment also for the SpanProcessor. Should I do this in this or separate PR?